okhttp/src/main/java/okhttp3/internal/CallEventListener.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2017 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3.internal;
+
+import okhttp3.Call;
+import okhttp3.EventListener;
+
+/** An {@link EventListener} scoped to a specific {@link Call}. */
+public final class CallEventListener {
```

Comment:
I suspect you can probably get away without this. Your fix of moving the code that creates the `EventListener` out of the constructor was sufficient – it’s still not concurrent and so I think we’re good.